File: LeetCode/Simulation/JS/Leetcode929-unique-email-addresses.js
var numUniqueEmails = function (emails) {
    let st = new Set();
    for (const s of emails) {
        let t = '';
        for (let i = 0; i < s.length; i++) {
            if (s[i] == '.') continue;
            if (s[i] == '@') {
                t += s.substr(i);
                break;
            }
            if (s[i] == '+') {
                while (s[i] != '@') i++;
                t += s.substr(i);
                break;
            }
            t += s[i];
        }
        st.add(t);
    }
    return st.size;
};

File: LeetCode/Simulation/Rust/Leetcode929-unique-email-addresses.rs
use std::collections::HashSet;
impl Solution {
  pub fn num_unique_emails(emails: Vec<String>) -> i32 {
    let mut st = HashSet::new();
    for s in &emails {
      let mut t = String::new();
      for mut i in 0..s.len() {
        if &s[i..i + 1] == "." {
          continue;
        }
        if &s[i..i + 1] == "@" {
          t += &s[i..s.len()];
          break;
        }
        if &s[i..i + 1] == "+" {
          while &s[i..i + 1] != "@" {
            i += 1;
          }
          t += &s[i..s.len()];
          break;
        }
        t += &s[i..i + 1];
      }
      st.insert(t);
    }
    st.len() as i32
  }
}
